pkg/policy/api/egress.go

```diff
@@ -104,6 +104,19 @@ type EgressCommonRule struct {
 	// +kubebuilder:validation:Optional
 	ToNodes []EndpointSelector `json:"toNodes,omitempty"`
 
+	// IPProtocols is a list of IP protocols that are allowed. This field is intended for protocols
+	// that operate directly over IP without transport-layer ports (unlike TCP or UDP).
+	//
+	// Example:
+	// - number: 2 #IGMP (Internet Group Management Protocol)
+	// - number: 112 #VRRP (Virtual Router Redundancy Protocol)
+	//
+	// These protocols do not use layer 4 transport port numbers and are identified solely
+	// by their IP protocol number (as defined in the IP header 'Protocol' field).
+	//
+	// +kubebuilder:validation:Optional
+	IPProtocols IPProtocols `json:"ipProtocols,omitempty"`
```

Comment:
I'm torn between adding a new field for maximum clarity, and relaxing the restriction that PortRule have a port number. Since we support port ranges, we already can express "allow all TCP packets" (albeit awkwardly). Why did you decide to add a new field?